source/extensions/quic_listeners/quiche/BUILD

```diff
@@ -381,8 +393,14 @@ envoy_cc_library(
 
 envoy_cc_library(
     name = "udp_gso_batch_writer_lib",
-    srcs = ["udp_gso_batch_writer.cc"],
-    hdrs = ["udp_gso_batch_writer.h"],
+    srcs = select({
```

Comment:
This requires a bunch of linux-only quic headers, so I had to exclude it from non-linux builds

Reply:
By doing so udp_gso_batch_writer_config_lib also needs to differentiate platforms I think.

Reply:
We wanted to make it so that configuring GSO on a non-supported platform wouldn't give the unhelpful error of "can't find an extension by this name", but instead gives the more specific error "Error configuring batch writer on platform without support for UDP GSO. Reset udp_writer_config to default writer". See #ifdef in `udp_gso_batch_writer_config.cc` that makes this happen.

Reply:
I see. Make sense!